torch/csrc/autograd/engine.h

```diff
@@ -69,9 +70,13 @@ struct TORCH_API Engine {
   std::once_flag start_threads_flag_;
   // Safe to read ready_queues_ without synchronization after intialization
   std::vector<std::shared_ptr<ReadyQueue>> ready_queues_;
+  // Number of reentrant threads currently accessing ready queue
+  std::vector<int> num_reentrant_threads_;
```

Comment:
Is this protected by a lock?

Reply:
The comment here would be clearer if it said something like, "For each ready queue, how many reentrant threads are currently accessing it."

Reply:
We only expect one thread to be accessing a particular ready at a time, so we don't need a lock.

Reply:
:+1: Worth a comment. Fun fact: accessing disjoint indices from different threads is thread safe... unless you are using`std::vector<bool>` (in which case it's not)